### DISCUSSION TITLE
Charger Power Approximation

### DISCUSSION THREAD

**Discussion by emaberga81** [alandtse/tesla#346] [Ideas]
Hi, I use the Charger Power value to calculate the charging cost. I nooticed that it is always approximated to an integer number. Would be great if it would be approximated with for example 2 decimals value so it would give a much better precise cost result.
Is this something that can be done?

**Comment by emaberga81** [comment]
anyone interested other than me? @alandtse maybe you could have a look at this?

**Reply by alandtse** [reply]
Please don't randomly ping me. The code is open source and we'd take the PR if you really want it.

**Comment by delbz** [comment]
Yes, this would help with my overall  energy management too. 
Even one decimal point would make a big difference. 
![D97DC320-DC49-43D3-9ED2-672957E80AAF](https://user-images.githubusercontent.com/34620121/209586692-78bc29a9-8e5a-408b-a340-ce95a3748a22.jpeg)

![24B2D9D2-12D6-4193-87BE-E1222D5AC9F4](https://user-images.githubusercontent.com/34620121/209586854-92d0287a-5e40-4a73-b664-cdb3b5d8cc9c.jpeg)

**Comment by InTheDaylight14** [comment]
Tesla API sends the charger_power without any decimal points. I suppose it can be changed to actual_current * charger_voltage / 1000

```
"charger_actual_current":12,
"charger_phases":1,
"charger_pilot_current":12,
"charger_power":1,
"charger_voltage":115
```

In my case the 5V drop between the plug and the car would not be captured but it could provide 1 more decimal point.

**Reply by delbz** [reply]
Yes, that would make it more accurate and help with overall home power management. 
Or if you could write us the code below for a template sensor to calculate it, that would be OK too. 😉

**Reply by InTheDaylight14** [reply]
```
{{ (state_attr('sensor.<vehicle_name>_charger_power', 'charger_amps_actual') * state_attr('sensor.<vehicle_name>_charger_power', 'charger_volts') / 1000) | round(1)}}
```
